### PR TITLE
Turn kStaticConstants to static_constants

### LINKED ISSUE #717: No kConstants
This hampers readability (`kIo`), we already don't do it everywhere (no `kMetre`, thankfully), this sort of constant is increasingly interchangeable with a function (cf. the introduction of variable templates, `SIUnit<T>()` could be a constant now).

###### Decision
Constants that would fall under the `kCamelCase` convention under the Google C++ styleguide should be named in `CamelCase` instead.

### PR DESCRIPTION
**Pull request by eggrobin**
Fix #717.

### CODE REVIEW

File: benchmarks/dynamic_frame.cpp [hunk 1]
```diff
@@ -222,10 +222,10 @@ void BM_BarycentricRotatingDynamicFrame(
   }
 }
 
-int const kIter = (1000 << 10) + 1;
+int const iter = (1000 << 10) + 1;
```
Comment (pleroy) [inline benchmarks/dynamic_frame.cpp]: iterations?  iteration_count?

File: benchmarks/quantities_body.hpp [hunk 2]
```diff
@@ -21,50 +21,50 @@ using si::Radian;
 
 namespace {
 
-std::size_t const kDimension = 100;
+std::size_t const dimension = 100;
```
Comment (pleroy) [inline benchmarks/quantities_body.hpp]: This identifier is weird and I have no idea what it means.  Can we find a better one?

File: geometry/epoch.hpp [hunk 3]
```diff
@@ -31,7 +31,7 @@ Instant const kJ2000;
 // Unix epoch, obtained by subtracting 30 years (including 7 leap years)
 // and the UTC time of day at J2000.0 from J2000.0.
 // +2000-01-01T00:00:00.000 (UTC)
-Instant const kUnixEpoch = kJ2000 - ((30 * 365 + 7) * Day + 11 * Hour +
+Instant const unix_epoch = kJ2000 - ((30 * 365 + 7) * Day + 11 * Hour +
```
Comment (pleroy) [inline geometry/epoch.hpp]: kJ2000 should be changed too.  Your regex is bad and you should feel bad.

File: integrators/sprk_integrator_body.hpp [hunk 4]
```diff
@@ -564,8 +564,8 @@ void SPRKIntegrator::SolveIncrement(
     Parameters<Position, Momentum> const& parameters,
     not_null<Solution<Position, Momentum>*> const solution) const {
   switch (vanishing_coefficients_) {
-    case kNone:
-      SolveIncrementOptimized<kNone, Position, Momentum>(
+    case none:
+      SolveIncrementOptimized<none, Position, Momentum>(
           compute_force, compute_velocity, parameters, solution);
       break;
     case kFirstBVanishes:
```
Comment (pleroy) [inline integrators/sprk_integrator_body.hpp]: Missed.

File: integrators/srkn_integrator.hpp [hunk 5]
```diff
@@ -74,7 +74,7 @@ class SRKNIntegrator : public MotionIntegrator {
 
  protected:
   enum VanishingCoefficients {
-    kNone,
+    none,
     kFirstBVanishes,
     kLastAVanishes,
```
Comment (pleroy) [inline integrators/srkn_integrator.hpp]: Missed.

File: ksp_plugin_test/interface_test.cpp [hunk 6]
```diff
@@ -172,28 +172,28 @@ TEST_F(InterfaceDeathTest, Errors) {
...
-    principia__UpdateCelestialHierarchy(plugin, kCelestialIndex, kParentIndex);
+    principia__UpdateCelestialHierarchy(plugin, celestial_index, parent_index);
   }, "plugin.*non NULL");
   EXPECT_DEATH({
-    principia__InsertOrKeepVessel(plugin, kVesselGUID, kParentIndex);
+    principia__InsertOrKeepVessel(plugin, kVesselGUID, parent_index);
   }, "plugin.*non NULL");
   EXPECT_DEATH({
     principia__SetVesselStateOffset(plugin,
                                     kVesselGUID,
```
Comment (pleroy) [inline ksp_plugin_test/interface_test.cpp]: Missed.

File: tools/generate_profiles.cpp [hunk 7]
```diff
@@ -13,7 +13,7 @@ namespace principia {
 namespace tools {
 
 namespace {
-const char kWarning[] = "// Warning!  This file was generated by running a "
+const char warning[] = "// Warning!  This file was generated by running a "
                         "program (see project |tools|).\n"
```
Comment (pleroy) [inline tools/generate_profiles.cpp]: Bad indent.

File: benchmarks/quantities_body.hpp (hunk 2 above)
Comment (eggrobin) [inline benchmarks/quantities_body.hpp]: Let's just kill this benchmark in another pull request.

File: tools/generate_profiles.cpp (hunk 7 above)
Comment (eggrobin) [inline tools/generate_profiles.cpp]: Done.

File: ksp_plugin_test/interface_test.cpp (hunk 6 above)
Comment (eggrobin) [inline ksp_plugin_test/interface_test.cpp]: Done.

File: integrators/srkn_integrator.hpp (hunk 5 above)
Comment (eggrobin) [inline integrators/srkn_integrator.hpp]: Done.

File: integrators/sprk_integrator_body.hpp (hunk 4 above)
Comment (eggrobin) [inline integrators/sprk_integrator_body.hpp]: Done.

File: geometry/epoch.hpp (hunk 3 above)
Comment (eggrobin) [inline geometry/epoch.hpp]: Done.

File: benchmarks/dynamic_frame.cpp (hunk 1 above)
Comment (eggrobin) [inline benchmarks/dynamic_frame.cpp]: Done.

**Pull request merged**